ratatui-core/src/layout/rect.rs

```diff
@@ -351,6 +353,61 @@ impl Rect {
         }
     }
 
+    /// Returns a new Rect, centered horizontally based on the provided constraint.
+    /// # Examples
```

Comment:
nit:

```suggestion
    /// Returns a new Rect, centered horizontally based on the provided constraint.
    ///
    /// # Examples
```

(applies for other doc comments as well)